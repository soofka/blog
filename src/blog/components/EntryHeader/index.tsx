import * as React from 'react';

import { TagInterface } from 'common/types';
import { EntryBackLink } from './components/EntryBackLink';
import { EntryHeaderMeta } from './components/EntryHeaderMeta';
import { EntryHeaderTags } from './components/EntryHeaderTags';
import { EntryHeaderTitle } from './components/EntryHeaderTitle';

interface EntryHeaderPropsInterface {
  title: string;
  titleLink: string;
  tags: TagInterface[];
  created: string;
  updated: string;
  full?: boolean;
}

export const EntryHeader = (props: EntryHeaderPropsInterface): JSX.Element => {
  const { title, titleLink, tags, created, updated, full } = props;

  return (
    <div>
      { full === true && <EntryBackLink/> }
      <EntryHeaderTitle
        title={title}
        link={titleLink}
      />
      <EntryHeaderMeta
        created={created}
        updated={updated}
      />
      <EntryHeaderTags
        tags={tags}
      />
    </div>
  );
};
