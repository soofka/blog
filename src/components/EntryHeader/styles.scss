@import 'components/Blog/variables.scss';

.entry-header {
  &--title {
    font-size: $fontSizeXL;
    margin: $marginS 0;
    padding: 0;

    a {
      color: $blue;
      text-decoration: none;

      &:hover {
        color: $lighterBlue;
      }
    }
  }

  &--meta {
    font-size: $fontSizeS;
    margin: $marginS $marginXXS $marginM $marginXXS;
    padding: 0;
    color: $lighterBlack;

    span {
      font-weight: bold;
    }
  }

  &--tags {
    font-size: $fontSizeS;
    margin: 0 0 $marginM 0;
    padding: 0;

    /* stylelint-disable */
    a {
    /* stylelint-enable */
      display: inline-block;
      margin: $marginXXS;
      padding: $paddingXS $paddingS;
      background-color: $darkWhite;
      color: $blue;
      text-decoration: none;

      &:hover {
        background-color: $darkerWhite;
        color: $darkBlue;
      }
    }
  }
}
