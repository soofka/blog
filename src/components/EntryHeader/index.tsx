import * as React from 'react';
import { Link } from 'react-router-dom';

import RoutingProvider from 'common/RoutingProvider';
import LabelsProvider from 'common/LabelsProvider';

import './styles.scss';

interface EntryHeaderPropsInterface {
  language: string;
  title: string;
  titleLink: string;
  tags: string[];
  created: string;
  updated: string;
}

export const EntryHeader = (props: EntryHeaderPropsInterface): JSX.Element => {
  const { language, title, titleLink, tags, created, updated } = props;

  return (
    <div className="entry-header">
      <h1 className="entry-header--title">
        <Link to={titleLink}>{title}</Link>
      </h1>
      <p className="entry-header--meta">
        {LabelsProvider.getLabel('entry__created', language)}
        <span>{created}</span>
        {LabelsProvider.getLabel('entry__last_updated', language)}
        <span>{updated}</span>
      </p>
      <p className="entry-header--tags">
        {tags.map((tag: string, index: number) => {
          return (
            <Link
              key={index}
              /* tslint:disable-next-line */
              to={`${RoutingProvider.getEntriesByTagNiceUrlBasePath()}/${RoutingProvider.parseTextToNiceUrl(tag)}`}
            >
              {tag}
            </Link>
          );
        })}
      </p>
    </div>
  );
};

export default EntryHeader;
