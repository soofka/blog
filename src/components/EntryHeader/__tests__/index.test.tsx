import * as React from 'react';
import { shallow } from 'enzyme';

import EntryHeader from 'components/EntryHeader';

describe('<EntryHeader/>', () => {

  it('renders correctly', () => {
    expect(shallow(
      <EntryHeader
        title="testTitle"
        titleLink="testTitleLink"
        tags={['tag1', 'tag2', 'tag3']}
        created="testCreated"
        updated="testUpdated"
      />)).toMatchSnapshot();
  });

});
