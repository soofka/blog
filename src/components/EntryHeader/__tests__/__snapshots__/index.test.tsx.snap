// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EntryHeader/> renders correctly 1`] = `
<div>
  <Component
    link="testTitleLink"
    title="testTitle"
  />
  <Component
    created="testCreated"
    updated="testUpdated"
  />
  <Component
    tags={
      Array [
        "tag1",
        "tag2",
        "tag3",
      ]
    }
  />
</div>
`;
